cmd/loki_exporter/loki_exporter.go

```diff
@@ -48,11 +49,16 @@ func main() {
 		log.Fatal(err)
 	}
 
-	prometheus.MustRegister(exporter)
-	prometheus.MustRegister(version.NewCollector("loki_exporter"))
+	// use our own registry to not get metrics from loki dependencies like etcd
+	r := prometheus.NewRegistry()
+	r.MustRegister(exporter)
+	r.MustRegister(version.NewCollector("loki_exporter"))
+	r.MustRegister(prometheus.NewGoCollector())
+	r.MustRegister(prometheus.NewProcessCollector(prometheus.ProcessCollectorOpts{}))
+	handler := promhttp.HandlerFor(r, promhttp.HandlerOpts{})
```

Comment:
Maybe you can add the [InstrumentMetricHandler](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus/promhttp?tab=doc#InstrumentMetricHandler):

```suggestion
	handler := promhttp.InstrumentMetricHandler(r, promhttp.HandlerFor(r, promhttp.HandlerOpts{}))
```

Reply:
Thanks for that, I added it.